Subtlety APL Update + Cleanup

syrifgit opened the pull request:
- Sync APL
- Whitespace, trailing commas
- Remove deprecated stuff (Marked for death, duplicate state expressions)
- Re-register gear under 1 call

syrifgit committed:
Outlaw Pack string

syrifgit committed:
Subtlety APL Update + Cleanup

- Sync APL
- Whitespace, trailing commas
- Remove deprecated stuff (Marked for death, duplicate state expressions)
- Re-register gear under 1 call

syrifgit commented:
Not me saying I replaced a variable with something a d then forgetting to actually do it. Will fix before reset tomorrow

Hekili commented on TheWarWithin/Priorities/RogueSubtlety.simc:
```diff
@@ -1,19 +1,21 @@
 actions.precombat+=/variable,name=priority_rotation,value=toggle.funnel
-actions.precombat+=/variable,name=trinket_sync_slot,value=1,if=trinket.1.has_stat.any_dps&(!trinket.2.has_stat.any_dps|trinket.1.is.treacherous_transmitter|trinket.1.cooldown.duration>=trinket.2.cooldown.duration)
-actions.precombat+=/variable,name=trinket_sync_slot,value=2,if=trinket.2.has_stat.any_dps&(!trinket.1.has_stat.any_dps|trinket.2.cooldown.duration>trinket.1.cooldown.duration)
+actions.precombat+=/variable,name=trinket_sync_slot,value=1,if=trinket.1.has_use_buff&(!trinket.2.has_use_buff|trinket.1.is.treacherous_transmitter|trinket.1.cooldown.duration>=trinket.2.cooldown.duration)
```
Not sure if this is an accurate revision (and it probably should get handled in the importer).

Looks like this should be supported via `mt_trinket_has_stat` though trinkets are a shitshow in terms of how the metatables are built.  See State.lua:547-578.

syrifgit replied:
Was changed on SimC, I guess they were having issues with it.

https://github.com/simulationcraft/simc/commit/4d78a56f38b6edff0babcd87862e72a9c5bc8279

Hekili commented on TheWarWithin/Priorities/RogueSubtlety.simc:
```diff
@@ -74,10 +79,10 @@ actions.finish+=/secret_technique,if=variable.secret
 actions.finish+=/rupture,cycle_targets=1,if=!variable.skip_rupture&!variable.priority_rotation&target.time_to_die>=(2*combo_points)&refreshable&active_enemies>=2
 # [Syrif] Uses Hekili-specific variable as a workaround
-actions.finish+=/rupture,if=talent.unseen_blade&variable.cooldowns_soon
+actions.finish+=/rupture,if=talent.unseen_blade&variable.cooldowns_soon&dot.rupture.remains<fight_remains
```
1. `fight_remains` references the death of the longest-lived enemy.  That means that when the current target has a 10 second Rupture, will die in 5 seconds, and another enemy will live for 20 seconds, `dot.rupture.remains<fight_remains` will be true and we'd refresh Rupture on a dying target.
2. There's a spec setting or something that suggests Rupture should only be used for enemies with at least X seconds to live.  If I'm not making that up, we probably want to determine how many seconds of Rupture we need to gain for this to truly be worthwhile.  Do we really want to re-Rupture when there are 4 seconds left on Rupture and the target will die in 5 seconds?  Seems like a wasted GCD vs. a direct damage finisher.

syrifgit replied:
This entry is already hacked together as-is, and the new condition is an extra **_restriction_** on top of what was there before

The purpose of the entry is "have a nice fat rupture up before cooldowns, to avoid using it during burst window". But this is the line that was causing the completely broken rotation with rupture spam early this season, so I had to build that `variable.cooldowns_soon` to manually keep it in line with sims. The issue was coming from the fact that "refreshable" wasn't really used here, because you don't want that restriction necessarily in this case. The sims would only do the entry once as written, but addon would spam, particularly if your CDs were held for any reason.

syrifgit committed:
Update RogueOutlaw.lua

syrifgit committed:
replace variable, fix symbols

syrifgit commented:
Finalized. Removed outlaw pack string (I forgot it was sitting on this branch, the `mt_set_bonus` deprecates it), replaced variables, added a quick fix for symbols of death spam in keys.
